src/tests/functional/shared_test_classes/src/base/utils/ranges.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/base/utils/ranges.hpp"
+
+#include <map>
+#include <queue>
+#include <vector>
+
+#include "common_test_utils/ov_tensor_utils.hpp"
+#include "gtest/gtest.h"
+#include "openvino/core/model.hpp"
+#include "openvino/core/node.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/core/type/element_type_traits.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "shared_test_classes/base/utils/generate_inputs.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+const std::shared_ptr<ov::test::utils::InputGenerateData> ModelRange::get_range_for_param(
+    const std::shared_ptr<ov::Node>& node) {
+    return node_ranges.at(get_range_id(node));
+}
+
+std::string ModelRange::get_range_id(const std::shared_ptr<ov::Node>& node) {
+    return node->get_name() + '_' + node->get_element_type().to_string();
+}
+
+ov::Tensor ModelRange::generate_input(std::shared_ptr<ov::Node> node, size_t port, const ov::Shape& targetShape) {
+    auto inputMap = ov::test::utils::getInputMap();
+    auto it = inputMap.find(node->get_type_info());
+    if (it == inputMap.end()) {
+        throw std::runtime_error("Couln't find Operation in inputMap: " + std::string(node->get_type_name()));
+    }
+
+    std::string range_id = get_range_id(node->get_input_node_shared_ptr(port));
+    return it->second(node, port, node->get_input_element_type(port), targetShape, node_ranges[range_id]);
+}
+
+void ModelRange::find_mode_ranges(const std::shared_ptr<ov::Model>& model) {
+    for (auto param : model->get_parameters()) {
+        std::shared_ptr<ov::test::utils::InputGenerateData> data =
+            std::make_shared<ov::test::utils::InputGenerateData>(ov::test::utils::rangeByType.get_range(param->get_element_type()));
+
+        bool range_corrected = true;
+        std::queue<std::shared_ptr<ov::Node>> queue;
+        queue.push(param);
```

Comment:
parameters? Because model can have many inputs

Reply:
Do you mean rename `queue` to `parameters` or add all parameters to `queue` ?
About second point, we integrate over parameters on str45